packages/webapp/pages/squads/[handle]/[token].tsx

```diff
@@ -123,44 +126,46 @@ const SquadReferral = ({
       event_name: AnalyticsEvent.ViewSquadInvitation,
       extra: joinSquadAnalyticsExtra(),
     });
+
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_type: TargetType.SquadJoinButton,
+      extra: JSON.stringify({
+        squad: member.source.id,
+        origin: Origin.SquadInvitation,
+        squad_type: member.source.public ? 'public' : 'private',
+      }),
+    });
+
```

Comment:
This should not live here right?
It's already handled inside the button which makes more sense.

Reply:
I see I think this is what I meant with duplicate code issue with splitting these buttons.
Can we find a way to uniform the behaviour?

It's pretty dangerous to have duplicated tracking events that do the exact same.
I'm not sure how much other stuff of this is duplicated?

Reply:
I agree this code duplication is highly error-prone. Ideally, we want to have one function/hook that triggers events. Duplicating is so risky, especially when it comes to tracking. I would suggest creating a hook to make sure that or a function. Happy with either of them

Reply:
Yeah agreed, made adjustments here a798c15530f4168858f031567b739efea55d54d7, decided to move tracking logic to button which is now reused on squad invitation page and open squad page.

Changes:
- added `SimpleSquadJoinButton` which contains base logic and tracking for button
- this one is used for `SquadJoinButton` which is the new one with join/leave capabilities needed for open squad
- `SimpleSquadJoinButton` is used on squad invite token page
- refactored layout of squad token page to not render button twice on mobile/desktop (would trigger events twice) - visually layout is the same